lib/datadog/tracing/configuration/http.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Tracing
+    module Configuration
+      module HTTP
+        # Datadog tracing supports capturing HTTP request and response headers as span tags.
+        #
+        # The provided configuration String for this feature has to be pre-processed to
+        # allow for ease of utilization by each HTTP integration.
+        #
+        # This class process configuration, stores the result, and provides methods to
+        # utilize this configuration.
+        class HeaderTags
+          # @param header_tags [Array<String>] The list of strings from DD_TRACE_HEADER_TAGS.
+          def initialize(header_tags)
+            @request_headers = {}
+            @response_headers = {}
+            @header_tags = header_tags || EMPTY
```

Comment:
After https://github.com/DataDog/dd-trace-rb/pull/2935/commits/a01cde6209eeeac9cc592284c879a1785b872708 we can get rid of `EMPTY` now 😄

Reply:
This suggestion kind of worked, but I need to fix a small thing in `Options` for it to work: https://github.com/DataDog/dd-trace-rb/pull/2994

Reply:
I rebased this PR on that fix, but everything here remains unchanged.